crates/bdk/src/wallet/mod.rs

```diff
@@ -792,36 +819,36 @@ impl<D> Wallet<D> {
     ///
     /// Note carefully that iterators go over **all** script pubkeys on the keychains (not what
     /// script pubkeys the wallet is storing internally).
-    pub fn spks_of_all_keychains(
+    pub fn all_unbounded_spk_iters(
         &self,
     ) -> BTreeMap<KeychainKind, impl Iterator<Item = (u32, ScriptBuf)> + Clone> {
-        self.indexed_graph.index.spks_of_all_keychains()
+        self.indexed_graph.index.all_unbounded_spk_iters()
     }
 
     /// Gets an iterator over all the script pubkeys in a single keychain.
     ///
-    /// See [`spks_of_all_keychains`] for more documentation
+    /// See [`all_unbounded_spk_iters`] for more documentation
     ///
-    /// [`spks_of_all_keychains`]: Self::spks_of_all_keychains
-    pub fn spks_of_keychain(
+    /// [`all_unbounded_spk_iters`]: Self::all_unbounded_spk_iters
+    pub fn unbounded_spk_iter(
```

Comment:
I don't think `all` is a better option. It can be interpreted as "all-revealed" or "all-stored". I'll be open for a better name, let me know if you have any further ideas.

Reply:
I like unbounded better. Although there is no "bounded" version it implies the others are bounded (which they are). "unbounded" tells the caller that it's never going to stop so it would only make sense to use this in a "stop gap" context.